include/roboteam_ai/analysis/PlaysObjects/Play.h

```diff
@@ -11,34 +11,51 @@
 
 namespace rtt::ai::analysis {
     /**
-     * The play has a vector of invariants, when the invariants are false the play is abandoned.
+     * The play has some invariants, defined in the isValidPlayToKeep function. When the invariants are false the play is abandoned.
      *
-     * TODO: Should this object have a behaviourtree associated to it? I think yes
      */
     class Play {
```

Comment:
I don't think a virtual abstract class is the way to deal with plays but it's definitely a maintainable OOP approach